src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -163,14 +163,12 @@ internal static FormulaValue TableFromJsonArray(JsonElement array, TableType tab
         {
             Contract.Assert(array.ValueKind == JsonValueKind.Array);
 
-            var records = new List<RecordValue>();            
-            bool isArray = tableType?._type.IsColumn == true;
-            FormulaType ft = isArray ? tableType.ToRecord().GetFieldType("Value") : tableType?.ToRecord();
+            var records = new List<RecordValue>();
 
             for (var i = 0; i < array.GetArrayLength(); ++i)
             {
                 JsonElement element = array[i];
-                var val = GuaranteeRecord(FromJson(element, ft, numberIsFloat));
+                var val = GuaranteeRecord(FromJson(element, tableType?.ToRecord(), numberIsFloat));
```

Comment:
>tableType?.ToRecord() [](http://example.com/codeflow?start=60&length=21)

Consider moving this call to outside the loop - the value will be the same for all elements.

Reply:
Done